src/devtools/views/Components/Element.js

```diff
@@ -149,10 +162,11 @@ export default function ElementView({ data, index, style }: Props) {
         marginBottom: `-${style.height}px`,
       }}
     >
+      <span className={styles.ScrollAnchor} ref={scrollAnchorStartRef} />
```

Comment:
If we're going to at least one extra `<span>` tag for each row,  why not just wrap one  around the toggle+name+$r group? Seems like maybe that would be a  simpler fix? (Unless I'm missing something?)

Reply:
The comment mentions this doesn't work. :-)

It's not sufficient because `scrollIntoView` doesn't actually guarantee its contents gets scrolled into view. (Yeah, wtf.) For some reason it's happy when some parts of the content are not in view — as long as some _are_ in view. I don't know what exact browser heuristics are. But that's the reason #142 is already happening.

By using 0px width we trick the browser to not apply these heuristics, and make the "is in view" decision "binary". Either it is, or it isn't. We do this for both sides, getting the desired behavior.

In fact you can see `scrollIntoView` alone isn't sufficient just from the "before" gif. Because we already have it around the component name — and yet you can see the component name being cut off a little all the time.

Reply:
Oh, I misunderstood the comment to mean that scrolling the whole row didn't work, because it's larger than the content. I thought a tightly-fit `<span>` _would_ work.  My apologies.

> It's not sufficient because scrollIntoView doesn't actually guarantee its contents gets scrolled into view.

Wtf indeed

Reply:
Browsers ¯\_(ツ)_/¯

Reply:
To clarify, because I'm still wtf-ing hard over here. The comment says:

> // However, even calling scrollIntoView() on their parent node
> // wouldn't guarantee that it will be *fully* brought into view.

Their "parent node" is literally the whole row:
<img width="1037" alt="Screen Shot 2019-04-12 at 6 22 52 PM" src="https://user-images.githubusercontent.com/29597/56072825-025e1d80-5d50-11e9-985c-6870e3e68e57.png">

I was thinking that a `<span>` wrapped around them (within the row, but not `width:100%`) would also fix this.

I assume you tried that and it didn't, but I just wanted to rule out a possible miscommunication 😄

Reply:
> In fact you can see scrollIntoView alone isn't sufficient just from the "before" gif. Because we already have it around the component name

I'm pretty sure this answers my question above

Reply:
Maybe I tried it wrong. (Yes, I tried it.)

This doesn’t rule out my original point though. Which is — if scrollIntoView guaranteed that a node fully gets into view, we wouldn’t observe the existing behavior where the component name *already doesn’t*.

So I’m not sure how even in theory wrapping them would help.

Reply:
Yeah, that bit popped back into my head after I'd left the comment. (I was on my way to the barber.) So I guess `scrollIntoView` is just...busted. Makes no sense though.